api/controllers/TagController.js

```diff
@@ -1,19 +1,90 @@
+var fetchAndPresentFollowed = (communityId, userId) =>
+  FollowedTag.where({community_id: communityId, user_id: userId})
+  .fetchAll({withRelated: 'tag'})
+  .then(followedTags =>
+    followedTags.map(followedTag => ({
+      name: followedTag.relations.tag.get('name')
+    })))
+
+var fetchAndPresentCreated = (communityId, userId) =>
+  CommunityTag.where({community_id: communityId, owner_id: userId})
+  .fetchAll({withRelated: 'tag'})
+  .then(createdTags =>
+    createdTags.map(createdTag => ({
+      name: createdTag.relations.tag.get('name')
+    })))
+
 module.exports = {
 
   findOne: function (req, res) {
     return Tag.find(req.param('tagName'))
     .then(tag => {
       if (!tag) return res.notFound()
       return CommunityTag.where({community_id: res.locals.community.id, tag_id: tag.id})
-      .fetch({withRelated: 'owner'})
+      .fetch({withRelated: [
+        'owner',
+        {'community.followedTags': qb => qb.where({'followed_tags.tag_id': tag.id, 'followed_tags.user_id': req.session.userId})}
+      ]})
       .then(communityTag => {
         var result = communityTag.pick('id', 'description', 'community_id')
         result.name = tag.get('name')
         result.owner = communityTag.relations.owner.pick('id', 'name', 'avatar_url')
+        result.followed = communityTag.relations.community.relations.followedTags.length > 0
+        result.created = result.owner.id === req.session.userId
         return result
       })
       .then(res.ok)
       .catch(res.serverError)
     })
+  },
+
+  findFollowed: function (req, res) {
+    return Community.find(req.param('communityId'))
+    .then(com => fetchAndPresentFollowed(com.id, req.session.userId))
+    .then(res.ok)
+    .catch(res.serverError)
+  },
+
+  findForLeftNav: function (req, res) {
+    return Community.find(req.param('communityId'))
+    .then(com => Promise.join(
+      fetchAndPresentFollowed(com.id, req.session.userId),
+      fetchAndPresentCreated(com.id, req.session.userId),
+      (followed, created) => ({
+        followed: _.filter(followed, f => !_.includes(_.map(created, 'name'), f.name)),
+        created: created
+      })
+    ))
+    .then(res.ok)
+    .catch(res.serverError)
+  },
+
+  follow: function (req, res) {
+    return Promise.join(
+      Tag.find(req.param('tagName')),
+      Community.find(req.param('communityId')),
```

Comment:
but req.param('communityId') is actually the slug, so I need to fetch for the actual id.
I called the param communityId by analogy with all the other routes that use it to mean id or slug.